singularity-eos/eos/eos_eospac.hpp

```diff
@@ -72,103 +72,1156 @@ class EOSPAC : public EosBase<EOSPAC> {
-  //                                         RealIndexer &&cvs,
-  //                                         const int num,
-  //                                         LambdaIndexer &&lambdas) const;
-  // template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  // inline
-  // void SpecificHeatFromDensityInternalEnergy(ConstRealIndexer &&rhos,
-  //                                            ConstRealIndexer &&sies,
-  //                                            RealIndexer &&cvs,
-  //                                            const int num,
-  //                                            LambdaIndexer &&lambdas) const;
-  // template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  // inline
-  // void BulkModulusFromDensityTemperature(ConstRealIndexer &&rhos,
-  //                                        ConstRealIndexer &&temperatures,
-  //                                        RealIndexer &&bmods,
-  //                                        const int num,
-  //                                        LambdaIndexer &&lambdas) const;
-  // template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  // inline
-  // void BulkModulusFromDensityInternalEnergy(ConstRealIndexer &&rhos,
-  //                                           ConstRealIndexer &&sies,
-  //                                           RealIndexer &&bmods,
-  //                                           const int num,
-  //                                           LambdaIndexer &&lambdas) const;
-
-  // template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  // inline
-  // void GruneisenParamFromDensityTemperature(ConstRealIndexer &&rhos,
-  //                                           ConstRealIndexer &&temperatures,
-  //                                           RealIndexer &&gm1s,
-  //                                           const int num,
-  //                                           LambdaIndexer &&lambdas) const;
-
-  // template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  // inline
-  // void GruneisenParamFromDensityInternalEnergy(ConstRealIndexer &&rhos,
-  //                                              ConstRealIndexer &&sies,
-  //                                              RealIndexer &&gm1s,
-  //                                              const int num,
-  //                                              LambdaIndexer &&lambdas) const;
-
-  // template<typename RealIndexer, typename LambdaIndexer>
-  // inline
-  // void FillEos(RealIndexer &&rhos, RealIndexer &&temps, RealIndexer &&energies,
-  //              RealIndexer &&presses, RealIndexer &&cvs, RealIndexer &&bmods,
-  //              const int num, const unsigned long output,
-  //              LambdaIndexer &&lambdas) const;
+  // EOSPAC vector implementations
+  template <int CHUNK_SIZE = 64, typename RealIndexer, typename ConstRealIndexer,
```

Comment:
I assume that if you want to use the `CHUNK_SIZE` template parameter, you need to use the `EOSPAC` class directly since this isn't exposed through the `Variant` class right? I don't think I'd advocate for any changes to the `Variant` class to accommodate this yet since it's a requirement specific to EOSPAC and unlikely to be needed by other EOS.

Reply:
Yes, that's the intention. It's a tradeoff between using more on-demand stack vs preallocating a huge heap scratch space based of `nscratch() * nelements * sizeof(Real)`. Which of the two implementations or if both should stay is still up for debate. I can imagine use cases for either one of them.